internal/controller/controlplane_controller.go

```diff
@@ -163,16 +166,38 @@ func (r *ControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.Request
 func (r *ControlPlaneReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&tenancyv1alpha1.ControlPlane{}).
-		Owns(&corev1.Service{}).
 		Owns(&networkingv1.Ingress{}).
 		Owns(&appsv1.Deployment{}).
 		Owns(&appsv1.StatefulSet{}).
```

Comment:
Same problem here too, if I understand https://github.com/kubestellar/kubeflex/blob/v0.7.2/pkg/reconcilers/vcluster/reconciler.go#L100 correctly. If the StatefulSet is missing, this controller will note that in the ControlPlane's status and happily be done, without getting notified when the StatefulSet is created later.